src/normalize/normalize.rs

```diff
@@ -0,0 +1,154 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::io::Error;
+use std::io::ErrorKind;
+use std::io::Result;
+use std::path::PathBuf;
+
+use crate::elf;
+use crate::elf::ElfParser;
+use crate::log;
+use crate::maps;
+use crate::maps::MapsEntry;
+use crate::maps::Pid;
+use crate::Addr;
+
+use super::meta::Binary;
+use super::meta::UserAddrMeta;
+
+
+/// A type capturing normalized addresses along with captured meta data.
+///
+/// This type enables "remote" symbolization. That is to say, it represents the
+/// input necessary for addresses to be symbolized on a system other than where
+/// they were recorded.
+#[derive(Clone, Debug)]
+pub struct NormalizedAddrs<M> {
+    /// Normalized addresses along with an index into `meta` for retrieval of
+    /// the corresponding [`AddrMeta`] information.
+    ///
+    /// A normalized address is one as it would appear in a binary or debug
+    /// symbol file, i.e., one excluding any relocations.
+    pub addrs: Vec<(Addr, usize)>,
+    /// Meta information about the normalized addresses.
+    pub meta: Vec<M>,
+}
+
+/// A type representing normalized user addresses.
+pub type NormalizedUserAddrs = NormalizedAddrs<UserAddrMeta>;
+
+
+/// Normalize a virtual address belonging to an ELF file represented by the
+/// provided [`MapsEntry`].
+fn normalize_elf_addr(virt_addr: Addr, entry: &MapsEntry) -> Result<Addr> {
+    let parser = ElfParser::open(&entry.path)?;
```

Comment:
As discussed in person, that's not really accurate. We already [strip](https://github.com/libbpf/blazesym/blob/master/src/maps.rs#L99-L103) the `(deleted)` prefix and fall back to using `map_files` then. I agree that we should use `map_files` throughout, but when I tried just stupidly switching over various things blew up and I have not had the time to look into it. So this will get done once I rectify that. But until then, I believe this logic should stay that way, as assumptions here are not different to those made elsewhere in the code.